tests/components/homekit/test_type_thermostats.py

```diff
@@ -2011,3 +2011,25 @@ async def test_thermostat_with_temp_clamps(hass, hk_driver, events):
     assert acc.char_target_heat_cool.value == 3
     assert acc.char_current_temp.value == 1000
     assert acc.char_display_units.value == 0
+
+
+async def test_thermostat_not_exist_hvac_action(hass, hk_driver, events):
+    """Test that unsupported HVAC action does not crash thermostat."""
+    entity_id = "climate.test"
+
+    hass.states.async_set(
+        entity_id,
+        HVAC_MODE_COOL,
+        {
+            ATTR_TEMPERATURE: 20.0,
+            ATTR_CURRENT_TEMPERATURE: 19.0,
+            ATTR_HVAC_ACTION: "None",
```

Comment:
If it's really getting quoted "None", this is a bug in the source data and probably shouldn't be changed in HomeKit.

Reply:
In any case, if a non-standard value appears, it will lead to an error and crash thermostat. I think it might be helpful to add an extra check.

Reply:
HVAC mode checked in as same way.
https://github.com/home-assistant/core/blob/f8a66ec1f2767c53553aeddb20ec9bc1c48cd1ed/homeassistant/components/homekit/type_thermostats.py#L453

Reply:
The integration is reasonable for validating data before setting it in state machine.  I'd consider it a bug in the integration that is injecting bad state data in the state machine since anything reading it is going to have a problem.   We shouldn't add workarounds to every integration that is consuming the data, we should fix it at the source.

If the value is passed in as `None` (no quotes), the climate entity won't set the property.
https://github.com/home-assistant/core/blob/dev/homeassistant/components/climate/__init__.py#L298

Reply:
In same time @Koenkk says: https://github.com/Koenkk/zigbee2mqtt/issues/9219#issuecomment-965659015

Reply:
I agree its a bug `handle_action_received` is lacking validation, and doesn't handle the `null`/`None` case.  It looks like its being converted to a string.
https://github.com/home-assistant/core/blob/dev/homeassistant/components/mqtt/climate.py#L407

`handle_swing_mode_received` does validation so it could be used as a template to fix it. to ensure that `hvac_action` is only a valid action as defined by the developer docs or `None` https://developers.home-assistant.io/docs/core/entity/climate/#hvac-action

Example
https://github.com/home-assistant/core/blob/dev/homeassistant/components/mqtt/climate.py#L509

Reply:
So, I understood you. How do we fix this bug? :)

Reply:
Modify `handle_action_received` in https://github.com/home-assistant/core/blob/dev/homeassistant/components/mqtt/climate.py#L407 to use the same type of validation that https://github.com/home-assistant/core/blob/dev/homeassistant/components/mqtt/climate.py#L509 uses

Reply:
Ohh, @bdraco Thanks a lot for the tip and review! 
P.S. If you still have time, please take a look at my HK [related issue](https://github.com/home-assistant/core/issues/59329).